src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -85,6 +86,38 @@ internal bool TryUnparse(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Defines the different type of stop words remover supported.
+        /// </summary>
+        public enum StopWordsRemoverType
+        {
+            /// <summary>
+            /// Use stop words remover that can removes language-specific list of stop words (most common words) already defined in the system.
+            /// </summary>
+            UsePredefinedStopWordsRemover,
+
+            /// <summary>
+            /// Use custom stop words remover that removes stop words based on custom defined list.
+            /// </summary>
+            UseCustomStopWordsRemover
+        }
+
+        /// <summary>
+        /// Defines the different type of stop words remover supported.
+        /// </summary>
+        public sealed class StopWordsRemoverOption
+        {
+            /// <summary>
+            /// Type of stop words remover to use.
+            /// </summary>
+            public StopWordsRemoverType StopWordsRemover;
+
+            /// <summary>
+            /// List of stop words. It is used when <see cref="StopWordsRemover"/> is equal to <see cref="StopWordsRemoverType.UseCustomStopWordsRemover"/>.
+            /// </summary>
+            public string[] StopWords;
```

Comment:
>string [](start = 19, length = 6)

Why not `IEnumerable<string>` ?

Reply:
I doubt that EntryPoint will break because of this. The actual interface is still using `string[]`
[https://github.com/dotnet/machinelearning/blob/91a8703c6e43078dc8516e6e7383a179f8ab5ea1/src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs#L643](https://github.com/dotnet/machinelearning/blob/91a8703c6e43078dc8516e6e7383a179f8ab5ea1/src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs#L643)

I have mentioned that in the comments in the my PR. I am waiting for the people to comment on it. So that I can decide what to do.

---
In reply to: [266099423](https://github.com/dotnet/machinelearning/pull/2962#discussion_r266099423) [](ancestors = 266099423)